docs/development-guide/extending-gaffer.md

```diff
@@ -0,0 +1,217 @@
+# Extending Gaffer
+
+Extending Gaffer can mean a few things to a user, this guide aims to cover
+general use cases for how and why you may want to extend the capabilities or add
+customisation to Gaffer.
+
+## Writing Custom Classes
+
+As Gaffer is Java based to create and load additional libraries you will first
+need to write your custom classes. Now these can be for a range of uses, Gaffer
+allows for many places to use custom classes such as, custom operations,
+aggregation functions, element generators etc.
+
+Depending on what type of class you are writing e.g. an `Operation` you may
+need extend or implement one of the interface classes already in Gaffer.
+Assuming you are using Maven for package management you can add Gaffer as
+a dependency to gain access to its classes like below:
+
+```xml
+<dependency>
+    <groupId>uk.gov.gchq.gaffer</groupId>
+    <artifactId>${gaffer.component}</artifactId>
+    <version>${gaffer.version}</version>
+</dependency>
+```
+
+!!! tip
+    See the [components breakdown](./project-structure/components/components.md)
+    for a list of key Gaffer modules.
+
+### Writing an Operation
+
+In Gaffer we use the term Operation to describe some component of a
+manipulation/query affecting a graph, these are chained together in `OperationChains`
+and this is how we execute manipulations/queries to our graphs. Whilst
```

Comment:
```suggestion
and this is how we execute queries to our graphs. Whilst
```
I don't think manipulations is usually used in the docs so seems strange to introduce a new term.